fontforge/parsepfa.c

```diff
@@ -2679,7 +2679,8 @@ void PSCharsFree(struct pschars *chrs) {
 return;
     for ( i=0; i<chrs->next; ++i ) {
 	if ( chrs->keys!=NULL ) free(chrs->keys[i]);
-	free(chrs->values[i]);
+	if ( chrs->lens!=NULL && chrs->lens[i]!=0 )
```

Comment:
Care to explain this further? How does the value of `lens` dictate whether or not `values` should be freed?

Reply:
Normally `values[i]` is either NULL (and therefore OK to free) or points to a chunk of memory of length `lens[i]`, as in the allocation in `findstring[]`. 

However, there is a special case in `findnumbers()` where `values[i]` is set with a `(void *)` cast directly to the output of `strtol()`. In this case `lens[i]` is helpfully set to `0`. 

The crash in #4592 is caused by attempting to free `0x...01` or `0x...02`.  (I don't remember which at this point.) So to me this looks like a special use case not getting a matching special deallocation case, although I did not go through every use of `lens` and `values` to determine exactly what's going on with the full subsystem.

Reply:
Oh, and I guess it's worth clarifying that line 1151 is the allocation of `values`, not `values[i]` and this PR changes the pattern of deallocation of `values[i]` only.